kyaml/fn/runtime/kubectl/kubectl.go

```diff
@@ -0,0 +1,263 @@
+	"sigs.k8s.io/kustomize/kyaml/fn/runtime/runtimeutil"
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// We're running krm via kubectl here:
+// 1. If PodTemplateName is set - taking it from kubectl (the same namespace), and using defaultPodTemplate overwise. See some mandatory fields values there, e.g. stdin.
+// 2. Creating pod using PodTemplate spec and applying imageUrl and set of envs
+// 3. Waiting maximum PodStartTimeout for the pod up and running
+// 4. using kubectl attach -q to send input and to get output
+// 5. deleting the created pod
+
+const (
+	defaultPodTemplate = `
+{
+    "apiVersion": "v1",
+    "kind": "PodTemplate",
+    "metadata": {
+        "name": "not-important"
+    },
+    "template": {
+        "spec": {
+            "containers": [
+                {
+                    "name": "default",
+                    "stdin": true,
+                    "stdinOnce": true
+                }
+            ],
+            "restartPolicy": "Never"
+	}
+    }
+}
+`
+	alphanums = "bcdfghjklmnpqrstvwxz2456789"
+)
+
+type Filter struct {
+	runtimeutil.ContainerSpec `json:",inline" yaml:",inline"`
+
+	PodTemplateName   string
+	KubectlGlobalArgs []string
+
+	PodStartTimeout *time.Duration
+
+	Exec runtimeexec.Filter
+}
+
+var rng *rand.Rand = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+func randString(n int) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = alphanums[rng.Intn(len(alphanums))]
+	}
+	return string(b)
+}
+
+func (c Filter) String() string {
+	if c.Exec.DeferFailure {
+		return fmt.Sprintf("%s deferFailure: %v", c.Image, c.Exec.DeferFailure)
+	}
+	return c.Image
+}
+func (c Filter) GetExit() error {
+	return c.Exec.GetExit()
+}
+
+func (c Filter) getPodTemplate() (*yaml.RNode, error) {
+	template := defaultPodTemplate
+	if c.PodTemplateName != "" {
+		//TODO: maybe we should cache it with timestamp... and update every 30s?
+		args := append(c.GetKubectlGlobalArgs(),
+			"get",
+			fmt.Sprintf("podTemplate/%s", c.PodTemplateName),
+			"-o",
+			"json")
+		out, err := osexec.Command("kubectl", args...).Output()
+		if err != nil {
+			return nil, fmt.Errorf("kubectl %v failed: %w", args, err)
+		}
+		template = string(out)
+	}
+
+	return yaml.ConvertJSONToYamlNode(template)
+}
+
+func (c Filter) getPodConfig(podName string) (string, error) {
+	template, err := c.getPodTemplate()
+	if err != nil {
+		return "", fmt.Errorf("getPodTemplate failed: %w", err)
+	}
+
+	res, err := yaml.Parse(
+		fmt.Sprintf(`
+apiVersion: v1
+kind: Pod
+metadata:
+  name: %s
+`,
+			podName,
+		),
+	)
+
+	spec, err := template.Pipe(yaml.PathGetter{Path: []string{"template", "spec"}})
+	if err != nil {
+		return "", fmt.Errorf("PathGetter for spec failed: %w", err)
+	}
+	if spec == nil {
+		return "", fmt.Errorf("specified template doesn't contain spec")
+	}
+	err = res.PipeE(yaml.FieldSetter{Name: "spec", Value: spec})
+	if err != nil {
+		return "", fmt.Errorf("FieldSetter for spec failed: %w", err)
+	}
+
```

Comment:
todo: copy labels if any from PodTemplate to Pod. They needed e.g. to restrict pod with NetworkPolicy that can be matched with labels